### PR TITLE
Enforce name separation

### PR DESCRIPTION
**Pull request by CyberiaResurrection**

### CODE REVIEW

File: src/POData/BaseService.php [hunk 1]
```diff
@@ -427,7 +428,8 @@ protected function serializeResult(RequestDescription $request, UriProcessor $ur
                     }
 
                     $odataModelInstance = $objectModelSerializer->writeUrlElement($result);
-                } elseif (TargetKind::RESOURCE() == $requestTargetKind) {
+                } elseif (TargetKind::RESOURCE() == $requestTargetKind
+                          || TargetKind::SINGLETON() == $requestTargetKind) {
```
Comment (CyberiaResurrection) [inline src/POData/BaseService.php:432]: Treating singletons as resources

File: src/POData/BaseService.php [hunk 2]
```diff
@@ -603,6 +606,7 @@ public function getResponseContentType(
                     )
                 );
 
+            case TargetKind::SINGLETON():
```
Comment (CyberiaResurrection) [inline src/POData/BaseService.php:609]: Again, treating singletons as resources - simpler, need to write less code, etc

File: src/POData/Providers/Metadata/ResourceFunctionType.php [hunk 3]
```diff
@@ -18,12 +18,15 @@ class ResourceFunctionType
...
+    private $resourceType = null;
+
     /**
      * ResourceFunctionType constructor.
      * @param string|array $functionName
      * @param FunctionImportAnonymousType $type
+     * @param ResourceType $resource
      */
-    public function __construct($functionName, FunctionImportAnonymousType $type)
+    public function __construct($functionName, FunctionImportAnonymousType $type, ResourceType $resource)
```
Comment (CyberiaResurrection) [inline src/POData/Providers/Metadata/ResourceFunctionType.php:29]: Need for this revealed by integration testing - ProvidersWrapper lost plot with null ref exception
